sdk/nodejs/resource.ts

```diff
@@ -1420,24 +1420,7 @@ export class ComponentResource<TData = any> extends Resource {
         remote: boolean = false,
         packageRef?: Promise<string | undefined>,
     ) {
-        // Explicitly ignore the props passed in.  We allow them for back compat reasons.  However,
-        // we explicitly do not want to pass them along to the engine.  The ComponentResource acts
-        // only as a container for other resources.  Another way to think about this is that a normal
-        // 'custom resource' corresponds to real piece of cloud infrastructure.  So, when it changes
-        // in some way, the cloud resource needs to be updated (and vice versa).  That is not true
-        // for a component resource.  The component is just used for organizational purposes and does
-        // not correspond to a real piece of cloud infrastructure.  As such, changes to it *itself*
-        // do not have any effect on the cloud side of things at all.
-        super(
-            type,
-            name,
-            /*custom:*/ false,
-            /*props:*/ remote || opts?.urn ? args : {},
-            opts,
-            remote,
-            false,
-            packageRef,
-        );
+        super(type, name, /*custom:*/ false, /*props:*/ args, opts, remote, false, packageRef);
         this.__remote = remote;
         this.__registered = remote || !!opts?.urn;
         this.__data = remote || opts?.urn ? Promise.resolve(<TData>{}) : this.initializeAndRegisterOutputs(args);
```

Comment:
Do we not need to change this to always take `args` as well?  I guess maybe it doesn't matter right now, because this is only used for stack outputs in https://github.com/pulumi/pulumi/blob/master/sdk/nodejs/runtime/stack.ts#L58 afaics.